zebra-consensus/src/transaction.rs

```diff
@@ -909,26 +909,26 @@ where
         let mut async_checks = AsyncChecks::new();
 
         if let Some(orchard_shielded_data) = orchard_shielded_data {
+            // # Consensus
+            //
+            // > The proof 𝜋 MUST be valid given a primary input (cv, rt^{Orchard},
+            // > nf, rk, cm_x, enableSpends, enableOutputs)
+            //
+            // https://zips.z.cash/protocol/protocol.pdf#actiondesc
+            //
+            // Queue the verification of the Halo2 proof for each Action
+            // description while adding the resulting future to our
+            // collection of async checks that (at a minimum) must pass for
+            // the transaction to verify.
```

Comment:
I think we don't queue the proof for each Action anymore. Maybe something like this?

```suggestion
            // Unlike Sapling, Orchard shielded transactions have a single 
            // aggregated Halo2 proof per transaction, even with multiple 
            // Actions in one transaction. So we queue it for verification 
            // only once instead of queuing it up for every Action description.
```